Add description about debugging profile

mochizuki875 opened the pull request:
This PR add description about debugging profile.

ref: 
https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/1441-kubectl-debug/README.md#debugging-profiles
https://github.com/kubernetes/kubernetes/pull/123149#discussion_r1598690150

ardaguclu reviewed (commented):
I dropped a few comments but overall sounds good to me. Thank you for fixing this.

mochizuki875 commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
```diff
@@ -639,3 +639,79 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+## Debugging Profiles {#debugging-profiles}
+
+When you use `kubectl debug` for debugging a node, a Pod by ephemeral container, or copy, you can apply a debugging profile to them via `--profile` flag.
+By applying a profile, specific properties such as [securityContexts](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) are set to them, allowing for adaptation to various scenarios.
```
@tengqm 

> Question: what does "them" refer to?

It indicates "a node, a Pod by ephemeral container, or copy"

tengqm replied:
Okay, that is a bit confusing to me.
If the "them" refers to the node, pod (copy) that is to be debugged,
then, the `securityContext` for "them" are adapted by applying a profile.
According to this statement, if 'baseline' profile is specified for a node debug session,
the `securityContext` of the Node is changed. I don't think that is correct.

mochizuki875 replied:
@tengqm 

Oh, I see!
I've rethought the sentence.
How about that.

tengqm replied:
Sorry I didn't see any changes related to the usage of "them".
There are two "them" in this paragraph.
Both of them are confusing.

mochizuki875 replied:
@tengqm 
I've fix that

> When you use `kubectl debug` for debugging a node, a Pod by ephemeral container, or copy, 

->
 
> When using `kubectl debug` to debug a node via a debugging Pod, a Pod via an ephemeral container, 
or a copied Pod,

By this change, I intended to indicate that each "them" refers to "a debugging Pod, a Pod via an ephemeral container, or a copied Pod" not Node itself.

> Both of them are confusing.

Should I explicitly indicate "Pod or container" instead of using "them"?

mochizuki875 replied:
I've rethought and removed second "them" as below:

> By applying a profile, specific properties such as [securityContext](/docs/tasks/configure-pod-container/security-context/)
are set, allowing for adaptation to various scenarios.

mochizuki875 commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
````diff
@@ -639,3 +639,79 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+{"privileged":true}
+```
+
+Clean up the Pod when you're finished with it:
````
@ardaguclu 

I've added that same as other section such as [Debugging with an ephemeral debug container](https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container), [Debugging using a copy of the Pod](https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#debugging-using-a-copy-of-the-pod) and [Debugging via a shell on the node ](https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#node-shell-session), but I'll remove that if it is not needed.

mochizuki875 commented:
@ardaguclu @tengqm 

Thank you for your feedback.
I've fixed them with leaving some replies to your comments.
How about that?

katcosgrove commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
```diff
@@ -639,3 +639,82 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+{{< /note >}}
+
+
+Assume that we create a Pod and debug it.
```
```suggestion
Assume that you create a Pod and debug it.
```

katcosgrove commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
````diff
@@ -639,3 +639,82 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+```
+
+Then, debug the Pod using an ephemeral container.
+If the ephemeral container needs to have privilege, we can use the `sysadmin` profile:
````
```suggestion
If the ephemeral container needs to have privilege, you can use the `sysadmin` profile:
```

katcosgrove commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
````diff
@@ -639,3 +639,82 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+/ #
+```
+
+Let's check the capabilities of the ephemeral container process by running the following command inside the container:
````
```suggestion
Check the capabilities of the ephemeral container process by running the following command inside the container:
```

katcosgrove commented on content/en/docs/tasks/debug/debug-application/debug-running-pod.md:
```diff
@@ -639,3 +639,82 @@ Don't forget to clean up the debugging Pod when you're finished with it:
+This means the container process is granted full capabilities as a privileged container by applying `sysadmin` profile.
+See more details about [capabilities](/docs/tasks/configure-pod-container/security-context/#set-capabilities-for-a-container).
+
+And we can also check that the ephemeral container was created as a privileged container:
```
```suggestion
You can also check that the ephemeral container was created as a privileged container:
```

katcosgrove reviewed (commented):
Quick pass for style guide compliance. Avoid using "we," and speak directly to the reader with "you."

mochizuki875 commented:
@katcosgrove 
Thank you for your comments.
I've fixed.

tengqm commented:
LGTM

katcosgrove commented:
Thank you!

/lgtm

mochizuki875 commented:
@ardaguclu 
Should I add a reference to this section [here](https://kubernetes.io/docs/tasks/debug/debug-cluster/kubectl-node-debug/)?

mochizuki875 commented:
OK, I've added reference to this section and rebased.

@tengqm @katcosgrove 
Sorry, could you check again and add `/lgtm` if there are no problems.

tengqm commented:
/lgtm

katcosgrove commented:
/approve
